android/src/main/java/com/sailthru/mobile/rnsdk/JsonConverter.kt

```diff
@@ -0,0 +1,126 @@
+package com.sailthru.mobile.rnsdk
+
+import androidx.annotation.VisibleForTesting
+import com.facebook.react.bridge.*
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Class for handling conversion between Native Maps and Arrays and JSON Objects and Arrays.
+ */
+internal class JsonConverter {
+
+    /**
+     * Convert JSONObject to WritableMap
+     */
+    @Throws(JSONException::class)
+    fun convertJsonToMap(jsonObject: JSONObject): WritableMap {
+        val writableMap: WritableMap = createNativeMap()
+        val iterator = jsonObject.keys()
```

Comment:
you could save some boilerplate code here with something like

```
    fun convertJsonToMap1(jsonObject: JSONObject) = createNativeMap().apply {
        for(key in jsonObject.keys()) {
            when (val value = jsonObject[key]) {
                is JSONObject -> putMap(key, convertJsonToMap(value))
                ...
                else -> putString(key, value.toString())
            }
        }
    }
```

similarly for the other iterator-based methods in this class